docs/docs/docs/01-core/admin/06-commands/01-build.md

```diff
@@ -0,0 +1,189 @@
+
+- miss-installed project
+- monorepos
+- incorrect/incompatible versions of packages for _certain_ packages like `styled-components` or `react`.
+
+The list of packages we explicity check for are:
+
+- `react`
+- `react-dom`
+- `styled-components`
+- `react-router-dom`
+
+This is because there should only be one instance of these packages installed and used by the project at any one time, failure to do so can and most likely will, lead to bugs. This also means an incompatible version of these packages could cause unintended side effects e.g. if `react@19` was suddenly released but we had not tested it against the admin panel.
+
+We run a prompt to encourage the user to install these deps – however, this functionality has not yet been built.
+
+### BuildContext
+
+The build context is the heart of how the admin builds, as said above it's agnostic, it doesn't care if we're using webpack or vite or parcel. It's an object of data that can be used to preapre any bundler. It's shape looks like:
+
+```ts
+interface BuildContext {
+  /**
+   * The absolute path to the app directory defined by the Strapi instance
+   */
+  appDir: string;
+  /**
+   * If a user is deploying the project under a nested public path, we use
+   * this path so all asset paths will be rewritten accordingly
+   */
+  basePath: string;
+  /**
+   * The current working directory
+   */
+  cwd: string;
+  /**
+   * The absolute path to the dist directory
+   */
+  distPath: string;
+  /**
+   * The relative path to the dist directory
+   */
+  distDir: string;
+  /**
+   * The absolute path to the entry file
+   */
+  entry: string;
+  /**
+   * The environment variables to be included in the JS bundle
+   */
+  env: Record<string, string>;
+  logger: Logger;
+  /**
+   * The build or develop options
+   */
+  options: Pick<BuildOptions, 'minify' | 'sourcemaps' | 'stats'> & Pick<DevelopOptions, 'open'>;
+  /**
+   * The plugins to be included in the JS bundle
+   * incl. internal plugins, third party plugins & local plugins
+   */
+  plugins: Array<{
+    path: string;
+    name: string;
+    importName: string;
+  }>;
+  /**
+   * The absolute path to the runtime directory
+   */
+  runtimeDir: string;
+  /**
+   * The Strapi instance
+   */
+  strapi: Strapi;
+  /**
+   * The browserslist target either loaded from the user's workspace or falling back to the default
+   */
+  target: string[];
+  tsconfig?: TSConfig;
+}
+```
+
+### Static Files
+
+The next step is to create a `runtime` folder in the root of the strapi project, a generic name `.strapi` is used and the build specifically uses a subfolder called `client`. This leaves more space for us to expand as and when we require it. We only generate two files for this – an `index.html` which is a static rendered React Component from the `@strapi/admin` package (DefaultDocument) & an `entry.js` file which calls the `renderAdmin` function & provides a mount point & plugin object.
```

Comment:
Will this change the primary output folder to something other than `build` within v4?

Reply:
No, I've specifically preserved the same behaviour either we take it from the tsconfig or it's build 😊

Reply:
Gotcha cool